builds/fabric8-che/plugins/keycloak-plugin-server/src/main/java/com/redhat/che/keycloak/server/KeycloakAuthenticationFilter.java

```diff
@@ -128,10 +135,14 @@ private boolean isWorkspaceAgentRequest(String authHeader) {
         return false;
     }
 
-    private boolean isWebsocketRequest(String requestURI, String requestScheme) {
-        return requestURI.endsWith("/ws") || requestURI.endsWith("/eventbus") || requestScheme.equals("ws")
-                || requestScheme.equals("wss") || requestURI.contains("/websocket/")
-                || requestURI.endsWith("/token/user");
+    private boolean isWsAgentUserTokenRequest(String requestURI) {
+        return requestURI.endsWith("/token/user");
+    }
+
+    private boolean isWebsocketRequest(String requestURI, String requestScheme, String upgradeHeader) {
+        return "websocket".equals(upgradeHeader) && //
```

Comment:
yes websocket is an upgrade of http
`//` is a trick to keep carriage return even if the formatter wouldn't do so
so it's to make the code more readable and avoid other people breaking the formatting chosen